### PR TITLE
  composition: check implementers of composed interfaces 

### PR DESCRIPTION
**Pull request by tomhoule**
The rule is that a composed interface is the union of all the interfaces
in all the schemas, and objects implementing that interface in the
composed schema must implement all fields of the interfface. In this PR,
we validate that.

closes GB-5507

### CODE REVIEW

File: engine/crates/composition/src/compose/enums.rs [hunk 1]
```diff
@@ -3,23 +3,22 @@ use std::collections::HashSet;
...
+pub(super) fn merge_enum_definitions<'a>(
+    first: &DefinitionWalker<'a>,
+    definitions: &[DefinitionWalker<'a>],
+    ctx: &mut Context<'a>,
 ) {
     let enum_name = first.name().id;
-    let is_inaccessible = definitions
-        .iter()
-        .any(|definition| definition.directives().inaccessible());
+    let directive_containers = definitions.iter().map(|def| def.directives());
```
Comment (Finistere) [inline engine/crates/composition/src/compose/enums.rs:12]: Not related to this PR, but why is it named `DirectiveContainerId`, why not `DirectiveId`? Does the `container` notion for a directive mean something specific?

File: engine/crates/composition/src/compose/enums.rs [hunk 2]
```diff
@@ -78,41 +77,58 @@ fn merge_intersection(
...
+        let name = all_values[start].0;
+        let end = all_values[start..].partition_point(|(n, _)| *n == name) + start;
+        let containers = all_values[start..end]
+            .iter()
+            .map(|(_, directives)| first.walk(*directives));
+        let composed_directives = collect_composed_directives(containers, ctx);
+
+        ctx.insert_enum_value(enum_id, first.walk(name), None, composed_directives);
+
+        if end == all_values.len() {
```
Comment (Finistere) [inline engine/crates/composition/src/compose/enums.rs:119]: `end` could be mutable and you would be able to do a simpler `while end < all_value.len()` loop

Comment (Finistere) [reply engine/crates/composition/src/compose/enums.rs:119]: or what you did later with mutable `start` :D

**Review by Finistere — APPROVED** [review]

File: engine/crates/composition/src/compose/enums.rs (hunk 1 above)
Comment (Finistere) [reply engine/crates/composition/src/compose/enums.rs:12]: I would use `DIrectiveLocationId` mapping to a list of directives from your description

Comment (tomhoule) [reply engine/crates/composition/src/compose/enums.rs:12]: It's something I introduced so we can handle directives uniformly across different kinds of schema items. It really payed off in ease of implementation of the last few composition features.

Comment (tomhoule) [reply engine/crates/composition/src/compose/enums.rs]: Renamed to DirectiveSite, even shorter.

**Pull request merged**